### PR TITLE
adapter: show cluster name and id in insights

### PR DESCRIPTION
**Pull request by maddyblue**
Although the text name is present in the explain plans, it's hard to parse, not in the JSON output, and doesn't include the cluster id.

### Motivation

  * This PR adds a feature that has not yet been specified.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - n/a

### CODE REVIEW

**Comment by maddyblue** [comment]
Not sure if someone from the cluster team thinks this should go elsewhere, like in one of the json plans as an analog to the text output. It wasn't obvious to me how to do that so I put it in an easier place instead.

**Comment by maddyblue** [comment]
fyi @SangJunBak

**Review by jkosh44 — APPROVED** [review]
LGTM
